hw/drivers/sensors/icp101xx/src/icp101xx.c

```diff
@@ -0,0 +1,692 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <string.h>
+#include <errno.h>
+#include <assert.h>
+
+#include "os/mynewt.h"
+#include "os/os_cputime.h"
+#include "hal/hal_i2c.h"
+#include "i2cn/i2cn.h"
+#include "sensor/sensor.h"
+#include "sensor/pressure.h"
+#include "sensor/temperature.h"
+#include "icp101xx/icp101xx.h"
+#include "icp101xx_priv.h"
+#include "modlog/modlog.h"
+#include "stats/stats.h"
+#include <syscfg/syscfg.h>
+
+/* Define the stats section and records */
+STATS_SECT_START(icp101xx_stat_section)
+    STATS_SECT_ENTRY(errors)
+STATS_SECT_END
+
+/* Define stat names for querying */
+STATS_NAME_START(icp101xx_stat_section)
+    STATS_NAME(icp101xx_stat_section, errors)
+STATS_NAME_END(icp101xx_stat_section)
+
+/* Global variable used to hold stats data */
+STATS_SECT_DECL(icp101xx_stat_section) g_icp101xxstats;
+
+#define ICP101XX_LOG(lvl_, ...) \
+    MODLOG_ ## lvl_(MYNEWT_VAL(ICP101XX_LOG_MODULE), __VA_ARGS__)
+
+/* Exports for the sensor API.*/
+static int icp101xx_sensor_read(struct sensor *, sensor_type_t,
+        sensor_data_func_t, void *, uint32_t);
+static int icp101xx_sensor_get_config(struct sensor *, sensor_type_t,
+        struct sensor_cfg *);
+
+static const struct sensor_driver g_icp101xx_sensor_driver = {
+    icp101xx_sensor_read,
+    icp101xx_sensor_get_config
```

Comment:
There is also a set_config function pointer which needs to get set to `config_icp101xx()` here.

Reply:
The icp101xx_sensor_set_config() has been added